src/irmin-pack/unix/lower.ml

```diff
@@ -264,4 +264,38 @@ module Make (Io : Io.S) (Errs : Io_errors.S with module Io = Io) = struct
 
   let read_exn ~off ~len ?volume t b =
     read_range_exn ~off ~min_len:len ~max_len:len ?volume t b
+
+  let create_copy ~src ~size lower_root =
+    let open Result_syntax in
+    let root = Layout.directory ~root:lower_root ~idx:0 in
+    let* () = Io.mkdir root in
+    let lower_data = Layout.data ~root in
+    let* () = Io.copy_file ~src ~dst:lower_data in
```

Comment:
I think we should use `Io.move_file` instead of `Io.copy_file`. One big reason is that copying the file will duplicate data on disk for a short period of time (before the call to `unlink`), which is a primary thing that what we want to avoid. 

But also, it's a lot slower than rename since it has to actually copy the file data. Eg

```
# fallocate -l 90G big.big (dd can trick you if you use zeros since it will create a sparse file and linux is smart :nerd_face:)

# time cp big.big big.big.big
^C (bc I was watching my disk be slowly eaten :hamburger:)

real    0m6.920s
user    0m0.008s
sys     0m6.814s

# time mv big.big big.big.big

real    0m0.005s
user    0m0.006s
sys     0m0.001s
```

Reply:
Yeah ok, aggreed! The plan was to go for a hardlink "copy" when possible, but let's revisit this later :)

Reply:
Aha, I was wondering what else you might be thinking with a "copy" -- that gives a better picture! :smile: Agreed to postpone fancier techniques to a later investigation. :pray: